torch_glow/src/PyTorchModelLoader.cpp

```diff
@@ -575,86 +581,213 @@ void PyTorchModelLoader::loadConstant(const torch::jit::Node *ptNode) {
     const auto &valsList = iVal.toBoolList();
     const auto &vals = c10::impl::toVector(valsList);
     if (vals.empty()) {
-      return;
+      return nullptr;
     }
     t.reset(glow::ElemKind::BoolTy, {vals.size()});
     auto handle = t.getHandle<bool>();
     for (size_t i = 0; i < vals.size(); ++i) {
       handle.raw(i) = vals[i];
     }
   } else if (iVal.isTensor()) {
-    assert(false && "not yet implemented tensor constants");
+    const at::Tensor &val = iVal.toTensor();
+    t = ptTensorToGlowTensor(val);
   } else if (iVal.isNone()) {
     // Nothing to do for None
-    return;
+    return nullptr;
   } else {
     assert(false && "Unsupported constant type");
   }
 
-  glow::Constant *glowConstant = mod->createConstant("constant", std::move(t));
+  return mod->createConstant("constant", std::move(t));
+}
+
+void PyTorchModelLoader::loadConstant(const torch::jit::Node *ptNode) {
+  auto inputs = ptNode->inputs();
+  auto outputs = ptNode->outputs();
+  assert(inputs.size() == 0);
+  assert(outputs.size() == 1);
+  (void)inputs;
+
+  auto optionalIValue = torch::jit::toIValue(outputs[0]);
+  assert(optionalIValue.has_value() && "Constants should have IValue outputs");
+  const torch::jit::IValue iVal = *optionalIValue;
+
+  glow::Constant *glowConstant = glowConstantFromIValue(iVal);
+
+  if (!glowConstant) {
+    return;
+  }
+
   addGlowNodeValue(outputs[0], glowConstant->getOutput());
 }
 
 void PyTorchModelLoader::populateNodeLoaderMapping() {
-  nodeLoaderMapping_[at::Symbol::fromQualString("prim::Constant")] =
-      [this](const torch::jit::Node *node) { return loadConstant(node); };
+  nodeLoaderMapping_.emplace(
+      at::Symbol::fromQualString("prim::Constant"),
+      NodeLoaderInfo({}, [this](const torch::jit::Node *node) {
+        return loadConstant(node);
+      }));
+
+  nodeLoaderMapping_.emplace(
+      at::Symbol::fromQualString("aten::mul"),
+      NodeLoaderInfo({0, 1}, [this](const torch::jit::Node *node) {
+        return loadMul(node);
+      }));
+  nodeLoaderMapping_.emplace(
+      at::Symbol::fromQualString("aten::mul_"),
+      NodeLoaderInfo({0, 1}, [this](const torch::jit::Node *node) {
```

Comment:
```suggestion
      NodeLoaderInfo({glow::AddNode::InputIndices::LHSIdx, glow::AddNode::InputIndices::RHSIdx}, [this](const torch::jit::Node *node) {
```
I doubt the mapping of indices to inputs will change, but in case it does, I think it would be better to refer to the indices using the enums from `AutogenNodes.h`.

Reply:
These aren't Glow inputs though

Reply:
Any suggestion on this are welcome. I could make each node loader a struct with an enum for inputs that get's used in loading and for this, but it seems like overkill

Reply:
Oh okay hmm...we have structs like these
```
  // Indexes of aten::_convolution inputs.
  struct Inputs {
    enum {
      input = 0, // NCHW
      weights = 1,
      bias = 2,
      stride = 3,
      padding = 4,
      dilation = 5,
      transposed = 6,
      output_padding = 7,
      groups = 8,
      benchmark = 9,
      deterministic = 10,
      cudnn_enabled = 11
    };
  };
```
defined the scope of loader functions already. I think we should pull them all out into a separate header.